src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Shape = 'triangle'| 'circle'| 'rectangle';
+type Color = 'red'| 'green'| 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const longestSide = [a, b, c].sort((side1, side2) => side2 - side1);
+
+    if (longestSide[0] >= (longestSide[1] + longestSide[2])) {
+      throw new Error('Check entered sides!');
+    } else if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Check entered sides!');
```

Comment:
You can improve error messages to be more descriptive about what exactly went wrong. For example, 'All sides must be greater than 0.'